jbi/router.py

```diff
@@ -75,15 +75,22 @@ def version(version_json: VersionDep):
     return version_json
 
 
-header_scheme = APIKeyHeader(name="X-Api-Key")
+header_scheme = APIKeyHeader(name="X-Api-Key", auto_error=False)
+basicauth_scheme = HTTPBasic(auto_error=False)
 
 
 def api_key_auth(
-    settings: SettingsDep, api_key: Annotated[str, Depends(header_scheme)]
+    settings: SettingsDep,
+    api_key: Annotated[str, Depends(header_scheme)],
+    basic_auth: Annotated[HTTPBasicCredentials, Depends(basicauth_scheme)],
 ):
-    if not secrets.compare_digest(api_key, settings.jbi_api_key):
+    if not api_key and basic_auth:
+        api_key = basic_auth.password
+    if not api_key or not secrets.compare_digest(api_key, settings.jbi_api_key):
         raise HTTPException(
-            status_code=status.HTTP_401_UNAUTHORIZED, detail="Forbidden"
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Incorrect API Key",
+            headers={"WWW-Authenticate": "Basic"},
```

Comment:
Nice catch. Along with the "Basic" challenge in the response, is there a good name for what we do with the `X-Api-Key` header?

Reply:
If we would read it from the `Authorization` header, we could add various `WWW-Authenticate` values, but as-is there's no way to declare that auth type to the client